.github/workflows/compatibility-tests.yml

```diff
@@ -6,18 +6,18 @@ on:
     branches:
       - master
   pull_request:
-    types: [labeled]
+    types: [ opened, synchronize, reopened, labeled ]
   workflow_dispatch:
 
 jobs:
   build:
-    if: ${{ github.event_name != 'labeled' || github.event.label.name == 'compatibility-sensitive' }}
+    if: github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'compatibility-sensitive')
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        hazelcast: [ '4.2.5', '5.0.3', '5.1.5', '5.2.1', '5.3.0-SNAPSHOT' ]
-        hibernate: [ '5.3.28.Final', '5.4.33.Final', '5.5.9.Final', '5.6.14.Final', '6.0.2.Final', '6.1.5.Final' ]
+        hazelcast: [ '5.0.4', '5.1.6', '5.2.3', '5.3.0','5.4.0-SNAPSHOT' ]
```

Comment:
Does 4.2.z have any particular issue? 4.2.z is out of support, but there are extended support contracts, if possible lets keep latest 4.2.z

Reply:
Re-added https://github.com/hazelcast/hazelcast-hibernate/pull/425/commits/19a13c784fce68c0e182643367b33fb268545ba3